File: GerenciadorDeCinema.WebApi/ViewModels/ModuloSessao/FormsSessaoViewModel.cs
using GerenciadorDeCinema.Dominio.Filmes;
using GerenciadorDeCinema.Dominio.ModuloSala;
using GerenciadorDeCinema.Dominio.ModuloSessao;

namespace GerenciadorDeCinema.WebApi.ViewModels.ModuloSessao
{
    public class FormsSessaoViewModel
    {
        Guid Id { get; set; }

        public DateTime Data { get; set; }

        public string HorarioInicio { get; set; }

        public decimal ValorIngresso { get; set; }

        public TipoAnimacao TipoAnimacao { get; set; }

        public TipoAudio TipoAudio { get; set; }

        public string FilmeId { get; set; }

        public string SalaId { get; set; }
    }

    public class InserirSessaoViewModel: FormsSessaoViewModel { }

    public class EditarSessaoViewModel : FormsSessaoViewModel { }

    public class VisualizarSessaoViewModel : FormsSessaoViewModel { }

    public class VisualizarSessaoCompletaViewModel : FormsSessaoViewModel
    {
        public string HorarioFim { get; set; }
    }
}

File: FrontEnd/src/app/sessoes/view-models/forms-sessao.view-model.ts
export class FormsSessaoViewModel {
  id: string;
  data: Date;
  horarioInicio: string;
  valorIngresso: string;
  tipoAnimacao: TipoAnimacaoEnum;
  tipoAudio: TipoAudioEnum;
  filmeId: string;
  salaId: string;
}

export enum TipoAnimacaoEnum {
  "2D",
  "3D"
}

export enum TipoAudioEnum {
  Original = 0,
  Dublado = 1
}
